### PR TITLE
Jr mh read write firestore

### LINKED ISSUE #1: 1. As a component, I want to be able to read from and write to the Firestore database so that users can persist information.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.

AC:

* The following have been added as project dependencies: `firebase` & `react-firebase-hooks`
* Make a change in the Firestore database and it shows up in the app
* Make a change in the app and it shows up in the Firestore database

Notes:

* Don't do too much here. In past, pairs have built the whole add item form in the first week. The best approach is just to make a throw away view with a `<button/>` and a `<div/>`. Click the `<button/>` to send some data to the database. Have the data show up in the `<div/>`. Simple!
* Also, if you need more direction on how to use `react-firebase-hooks`, you can reference this commit from Alejandro: https://github.com/alejandronanez/smart-shopping-list-demo/commit/d590adc3224872e71d000c41977a1b7a84542c31 and/or watch this Loom video where he explains how it works: https://www.loom.com/share/0dffa1a63bf24b449d6565c76606bbd9

### PR DESCRIPTION
**Pull request by JordanRaleigh**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

This PR adds a single component "ReadWriteFirestore.js" that renders a button and all information currently in the collection. The button when clicked, adds +1 to the state and adds the current state to the collection. The useEffect will then listen for updates on the collection and when updated, will map over the collection and display all values. 

## Related Issue

Closes #1 

## Acceptance Criteria

- [x] The following have been added as project dependencies: firebase //& react-firebase-hooks
- [x] Make a change in the Firestore database and it shows up in the app
- [x] Make a change in the app and it shows up in the Firestore database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

1. Open project and new terminal 
2. npm i
2. npm start to run server
3. Go to browser http://localhost:3000/
3. Click the button and see the collection render out

### CODE REVIEW

**Review by jonathanyeong — APPROVED** [review]
This looks awesome! I like how clean the code is. And well done with working out the new Firestore API!! 👏 👏 👏

**Review by timamero — APPROVED** [review]
Code looks great! Easy to read and understand. 
I also want to note that when I was testing the app I got the same firebase errors mentioned by Shalom.

File: src/components/ReadWriteFirestore.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import React, { useState, useEffect } from 'react';
+import { collection, addDoc, getDocs } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+
+function ReadWriteFirestore() {
+  const [state, setState] = useState(0);
+  const [user, setUser] = useState([]);
```
Comment (abowler2) [inline src/components/ReadWriteFirestore.js:7]: As you likely know, 'naming is hard' ™️  but it also allows you to convey a lot of meaning/intention when coding. In this instance, I found myself a little lost at first reading the code as this app will be generating a `List` of `items` so the use of `user` through me off a bit.

This is by no means a stopper for this PR and can definitely be addressed in later work as you build out the App.

File: src/components/ReadWriteFirestore.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  const handleSubmission = async () => {
+    setState(state + 1);
+    const addUser = await addDoc(collection(db, 'users'), {
```
Comment (abowler2) [inline src/components/ReadWriteFirestore.js:11]: No need to cast this to a variable as you are not actually using the variable.

**Comment by jonathanyeong** [comment]
@Mandihamza Now that you have reviews from both the other pair and mentors - feel free to merge!!

**Pull request merged**